api/language-extensions/syntax-highlight-guide.md

```diff
@@ -325,4 +315,209 @@ If you can't add a `meta.embedded.*` scope to your grammar, you can alternativel
 }
 ```
 
+## Semantic Tokenization
+
+Semantic tokenization has been added a new feature in 1.44. It allows language servers to provide additional token information based on the language server's knowledge on symbols in the source file. Themes can opt-in to use that information to improve and refine the syntax highlighting based on the TextMate grammar. The editor applies the highlighting that comes from semantic tokens on top of the highlighting from syntax tokens.
+.
+To do so, language extensions can register a `semantic token provider` by document language and/or file name. The editor will make requests to the providers when semantic tokens are needed.
+
+```ts
+const tokenTypes = ['class', 'interface', 'enum', 'function', 'variable'];
+const tokenModifiers = ['declaration', 'documentation'];
+const legend = new vscode.SemanticTokensLegend(tokenTypes, tokenModifiers);
+
+const provider: vscode.DocumentSemanticTokensProvider = {
+  provideDocumentSemanticTokens(document: vscode.TextDocument): vscode.ProviderResult<vscode.SemanticTokens> {
+    // analyze the document and return semantic tokens
+  }
+};
```

Comment:
We should definitely have a sample where user can just clone and run.
The doc should link to it in the beginning.